### PR TITLE
UDP engine aborts on networking-related errors from socket syscalls #2862

### PR DESCRIPTION
**Pull request by atomashpolskiy**
This refactoring is supposed to prevent ZeroMQ from terminating upon UDP engine tuning errors (in `zmq::udp_engine_t::plug`). As far as I was able to figure out, the session will terminate silently on `protocol_error`, which isn't good, probably. Not really sure, where to go from here...

### CODE REVIEW

File: src/tcp.cpp [hunk 1]
```diff
@@ -318,46 +318,7 @@ int zmq::tcp_read (fd_t s_, void *data_, size_t size_)
 
 void zmq::tcp_assert_tuning_error (zmq::fd_t s_, int rc_)
```
Comment (bluca) [inline src/tcp.cpp]: why keeping this function, if the usage changes to the new one from ip.cpp?

**Comment by bluca** [comment]
please add a relicense statement, and fix the formatting error that the CI is showing

File: src/tcp.cpp (hunk 1 above)
Comment (atomashpolskiy) [reply src/tcp.cpp]: It's public, so I was not sure, if this function might have been used somewhere outside libzqm. Removed it completely.

Comment (sigiesec) [reply src/tcp.cpp]: Only the API in zmq.h is exposed to outside of libzmq

**Comment by atomashpolskiy** [comment]
> please add a relicense statement, and fix the formatting error that the CI is showing

I had fixed the formatting of udp_engine.cpp in fe2287d (by `clang-format -i -style=file src/udp_engine.cpp`), so I'm not sure, what's up with Travis... Let's wait for the next build to finish and see, if the issue pertains

**Comment by bluca** [comment]
```make clang-format-diff``` will run it with the template we checked in

**Pull request merged**